pkg/networkserver/grpc_gsns.go

```diff
@@ -546,94 +632,104 @@ func (ns *NetworkServer) handleUplink(ctx context.Context, up *ttnpb.UplinkMessa
 			}
 			up.Settings.DataRateIndex = upDRIdx
 
+			var lastAt time.Time
+			nbTrans, lastAt = transmissionNumber(up.RawPayload[:len(up.RawPayload)-4], stored.RecentUplinks...)
+			if nbTrans > stored.MACState.CurrentParameters.ADRNbTrans ||
+				!lastAt.IsZero() && up.ReceivedAt.Sub(lastAt) > stored.MACState.CurrentParameters.Rx1Delay.Duration()+time.Second+maxTransmissionDelay {
+				logger.Warn("A more recent uplink retransmission was received during uplink handling, drop")
```

Comment:
I don't understand this message. Isn't this simply a retransmission that's sent too often or too late, i.e. the possible replay attack?

Also I don't know if this should be a warning in our logs. Don't we emit events for dropped traffic?

Note that logs are for ops. If there are warnings, there are things to look into operationally.

Reply:
This means that the delay between the matching and the transaction is too high, which is indeed an for ops. This is a NS/database performance issue.